### PR TITLE
Added one target calibration

### PR DESCRIPTION
**Pull request by marc-tonsen**
Added a new calibration routine: A single gaze target is displayed statically on the screen, the user gazes onto the target while moving his head to quickly sample many different gaze angles. 
With this routine one can quickly sample a great range densely.

### CODE REVIEW

File: pupil_src/shared_modules/calibration_routines/one_target_calibration.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+def on_resize(window,w,h):
+    active_window = glfwGetCurrentContext()
+    glfwMakeContextCurrent(window)
+    adjust_gl_view(w,h)
+    glfwMakeContextCurrent(active_window)
+
+
+
+class One_Target_Calibration(Calibration_Plugin):
+    """Calibrate using a marker on your screen
```
Comment (mkassner) [inline pupil_src/shared_modules/calibration_routines/one_target_calibration.py]: please update this.

**Comment by mkassner** [comment]
Final touch: think about the name? maybe single_marker calibration? or stationary_marker_calibration?

**Comment by cpicanco** [comment]
+1 stationary_marker_calibration
+2 static_marker_calibration

**Comment by willpatera** [comment]
The word `stationary` could be more easily misinterpreted IMO. I would opt for the word `static` if we only have a few choices. However, both of these terms could be misleading to the first-time user. 

This calibration procedure might be better described by what moves instead of what is not moving. The marker is fixed in place. The participant fixes their gaze on the marker center throughout the duration of the calibration. The participant _rotates_ their head while fixating on the marker center to collect samples within their FOV. 

Here is another alternative that is participant-centric: `head_rotation_calibration` or `neck_rotation_calibration` - "Look at the marker center, rotate your head while looking at the marker center to cover your FOV."

**Comment by cpicanco** [comment]
Will, your naming suggestion is not consistent with existing calibration
names. Calibration routines are being described by stuff in the environment
and presentation styles, not by eye movements. Regardless of what name will
be chosen, I would suggest to keep consistency with existing names. For
users, pupil docs will describe the procedure in detail. For programmers,
the doc section will also describe further details and so avoid any
misleading name only feature.

On Jun 2, 2017 00:14, "Will" <notifications@github.com> wrote:

The word stationary could be more easily misinterpreted IMO. I would opt
for the word static if we only have a few choices. However, both of these
terms could be misleading to the first-time user.

This calibration procedure might be better described by what moves instead
of what is not moving. The marker is fixed in place. The participant fixes
their gaze on the marker center throughout the duration of the calibration.
The participant *rotates* their head while fixating on the marker center to
collect samples within their FOV.

Here is another alternative that is participant-centric:
head_rotation_calibration or neck_rotation_calibration - "Look at the
marker center, rotate your head while looking at the marker center to cover
your FOV."

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/pupil-labs/pupil/pull/732#issuecomment-305678151>, or mute
the thread

**Comment by mkassner** [comment]
OK. lets conclude on this:

single_marker_calibration
one_marker_calibration
...

**Comment by willpatera** [comment]
@mkassner I vote for `single_marker_calibration`

**Comment by papr** [comment]
+1 on `single*`

**Comment by mkassner** [comment]
@marc-tonsen  please change the plugin name and I will merge this PR.

**Pull request merged**